### ISSUE TITLE
PS1.7.6.1 Quantity discounts not shown for default variant

### ISSUE THREAD

**Issue by ziezo** [PrestaShop/PrestaShop#16417]
The product page /{id_product}-xxx shows the default variant dropdowns but does not show the quantity discounts for the default variant

Only after selecting another variant or changing quantity the quantity discounts are shown, this is after the browser redirects to the variant page /{id_product}-{id_product_attribute}-xxx

[tested on a fresh PS1.7.6.1 instance]
**Steps to reproduce the issue**
1) install PS1.7.6.1
2) add quantity discount for product 3 "The best is yet to come' Framed poster" for combination "40x60"
3) Click on the Preview button => browse to http ://xxx.xxx/en/art/3-the-best-is-yet-to-come-framed-poster.html
table quantity discount prices for "40x60" is NOT shown
4) browse to http ://xxx.xxx/en/art/3-13-the-best-is-yet-to-come-framed-poster.html#/19-dimension-40x60cm
table quantity discount prices for "40x60" is shown

**Comment by khouloudbelguith** [comment]
Hi @ziezo,

I did not manage to reproduce the issue with PS1.7.6.1.
I attached a screen record.
https://drive.google.com/file/d/1IoEBkU6GT2qGx5Nhm2cDeNWc-uGGF1fF/view
Thanks to check & feedback.

**Comment by ziezo** [comment]
Steps to reproduce:
1) install PS1.7.6.1
2) add quantity discount for product 3 "The best is yet to come' Framed poster" for combination "40x60"
3) browse to http ://xxx.xxx/en/art/3-the-best-is-yet-to-come-framed-poster.html
table quantity discount prices for "40x60" is NOT shown
4) browse to http ://xxx.xxx/en/art/3-13-the-best-is-yet-to-come-framed-poster.html#/19-dimension-40x60cm
table quantity discount prices for "40x60" is shown

## NO QUANTITY PRICES http ://xxx.xxx/en/art/3-xxx
![image](https://user-images.githubusercontent.com/36749330/68878691-6e4e9d00-0708-11ea-8797-fba75e0fd066.png)

## QUANTITY PRICES http ://xxx.xxx/en/art/3-13-xxx
![image](https://user-images.githubusercontent.com/36749330/68878522-29c30180-0708-11ea-9a8a-41772e7691b7.png)

## Quantity discount setup
![image](https://user-images.githubusercontent.com/36749330/68877962-2f6c1780-0707-11ea-9486-3766e4186f75.png)

**Comment by khouloudbelguith** [comment]
Hi @ziezo,

Thanks for these clarifications.
I manage to reproduce the issue with PS1.7.6.1 & PS1.7.6.2build1.
When I click on the preview button, we are redirected to this URL: `http://shop.com/category/id_product-friendlyUrl `=> so the table quantity discount for the combination by default is not displayed.

It is not the same behavior with PS1.7.5.2, when we click on the preview button, we are redirected to this URL: `http://shop.com/category/id_product-id_combination-friendlyUrl ` => so the table quantity is displayed.

I’ll add this to the debug roadmap so that it’s fixed. If you have already fixed it on your end or if you think you can do it, please do send us a pull request!
Thanks!

**Comment by ziezo** [comment]
Thanks @khouloudbelguith !

This bug probably also affects other things related to id_product_attribute - such as price, but I did not test this.

I patched ProductController.php as follows. (Next time I'll try a pull request, still have to learn how to do that...)
```
    /**
     * Return id_product_attribute by id_product_attribute request parameter
     * or by the group request parameter.
     *
     * @return int|null
     *
     * @throws PrestaShopException
     */
    private function getIdProductAttributeByRequestOrGroup()
    {
        $requestedIdProductAttribute = (int) Tools::getValue('id_product_attribute');

        $groupIdProductAttribute = $this->getIdProductAttributeByGroup();
        $requestedIdProductAttribute = null !== $groupIdProductAttribute ? $groupIdProductAttribute : $requestedIdProductAttribute;

//BEGIN PATCH
        //return $this->tryToGetAvailableIdProductAttribute($requestedIdProductAttribute);
        $id_product_attribute = $this->tryToGetAvailableIdProductAttribute($requestedIdProductAttribute);
        if(!$id_product_attribute) $id_product_attribute = $this->product->getDefaultAttribute($this->product->id);
        return $id_product_attribute;
//END PATCH
   }
```

**Comment by Progi1984** [comment]
Merged in #16423
